backend/core/src/chain.rs

```diff
@@ -257,29 +255,41 @@ impl Message for GetNodeNetworkState {
     type Result = Option<Bytes>;
 }
 
+impl NodeSource {
+    pub fn init(self, nid: NodeId, chain: Addr<Chain>) -> bool {
+        match self {
+            NodeSource::Direct { conn_id, node_connector } => {
+                node_connector
+                    .try_send(Initialize {
+                        nid,
+                        conn_id,
+                        chain,
+                    })
+                    .is_err()
+            },
+            NodeSource::Shard => {
+                // TODO
+                false
+            }
+        }
+    }
+}
+
 impl Handler<AddNode> for Chain {
     type Result = ();
 
     fn handle(&mut self, msg: AddNode, ctx: &mut Self::Context) {
```

Comment:
Maybe I missed it, but where do we check if a node was already connected `Direct`/`Shard` and then tries to open a second connection?

Reply:
We don't, there is nothing stopping anyone from spawning a bunch of telemetry connections to the server (doesn't matter if they go through shards or directly).

Reply:
Ok, that's probably fine as long as nobody's trying to dos the server, which is not an entirely unlikely thing to happen. Hmm.